src/java/picard/sam/revertsam/OutputMap.java

```diff
@@ -0,0 +1,82 @@
+package picard.sam.revertsam;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import picard.PicardException;
+import picard.util.TabbedTextFileWithHeaderParser;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents the mapping from each read group id to the corresponding output file
+ * that RevertSam is expected to write to.
+ */
+public class OutputMap {
+    final Map<String, File> idToOutput = new HashMap<String, File>();
+
+    OutputMap(final File outputMapFile) {
```

Comment:
Why not simplify and have two methods that return a `Map<String,File>` that stores for each read group id the corresponding paths?